commit a39175fcde7b7ed59fd48b8cdfbd1ea63fb8e2aa (HEAD -> master)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2019-09-17T12:25:52-04:00 (3 weeks ago)
    
    MANTA-4514 incorrect Next-Marker response from ListBucketObjects when page ends on a delimiter group
    Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
    Approved by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index 2454f13..f35dced 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -882,8 +882,10 @@ function listBuckets(options) {
             assert.string(record.name, 'record.name');
 
             if (record.type === 'group') {
+                assert.optionalString(record.nextMarker, 'record.nextMarker');
                 obj = {
                     name: record.name,
+                    nextMarker: record.nextMarker,
                     type: 'group'
                 };
 
@@ -1356,8 +1358,10 @@ function listObjects(options) {
             assert.string(record.name, 'record.name');
 
             if (record.type === 'group') {
+                assert.optionalString(record.nextMarker, 'record.nextMarker');
                 obj = {
                     name: record.name,
+                    nextMarker: record.nextMarker,
                     type: 'group'
                 };
 
@@ -1533,6 +1537,7 @@ function paginationStream(opts, onRecord, done) {
         // send the group record
         sendRecord({
             name: base + delimiter,
+            nextMarker: nextMarker,
             type: 'group'
         });
 
